irods/test/scripts/test007_pam_features_in_new_auth_framework.bats

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bats
+#
+# Test creation of .irodsA for iRODS pam_password authentication, this time purely internal to the PRC
+# library code.
+
+. "$BATS_TEST_DIRNAME"/test_support_functions
+PYTHON=python3
+
+# Setup/prerequisites are same as for login_auth_test.
+# Run as ubuntu user with sudo; python_irodsclient must be installed (in either ~/.local or a virtualenv)
+#
+
+ALICES_ORIGINAL_PAM_PASSWORD=test123
+ALICES_NEW_PASSWORD=test_1234
+
+setup()
+{
+    export SKIP_IINIT_FOR_PASSWORD=1
+    setup_pam_login_for_alice "$ALICES_ORIGINAL_PAM_PASSWORD"
+    unset SKIP_IINIT_FOR_PASSWORD
+}
+
+teardown()
+{
+#    finalize_pam_login_for_alice
+#    test_specific_cleanup
+  :
+}
+
+@test f001 {
+
+    local AUTH_FILE=~/.irods/.irodsA
+
+    # Test assertion: No pre-existing authentication file.
+    [ ! -e $AUTH_FILE ]
+
+    python -c "import sys, irods.client_init; \
+               pw=sys.stdin.readline().strip(); irods.client_init.write_pam_irodsA_file(pw)" \
+              <<<"$ALICES_ORIGINAL_PAM_PASSWORD"
+    [ -e $AUTH_FILE ]
+
+    # ===  test we can log in.
+
+    local SCRIPT="
+import irods.test.helpers as h
+ses = h.make_session()
```

Comment:
Is `h.make_session()` doing something special to make the authentication happen?

Reply:
`make_session( )` is like  calling  `irods.session.iRODSSession`  surrounded by other checks and such. `make_session` is mostly intended as a shorthand for the purpose of writing test scripts.

Reply:
Got it.